java/api-test/api-usage-example/src/main/java/InlineFunctionExample.java

```diff
@@ -20,107 +19,127 @@
-                    Demo was evidently Greek, as her name, a traditional epithet of Demeter,
-                     signifies. The name was relatively common in the Hellenistic world, in
-                     Egypt and elsewhere, and she cannot be further identified. The date of her
-                     visit to the Colossus of Memnon cannot be established with certainty, but
-                     internal evidence on the left leg suggests her poem was inscribed there at
-                     some point in or after AD 196.[2]
-                    Epigram
-                    There are a number of graffiti inscriptions on the Colossus of Memnon.
-                     Following three epigrams by Julia Balbilla, a fourth epigram, in elegiac
-                     couplets, entitled and presumably authored by "Demo" or "Damo" (the
-                     Greek inscription is difficult to read), is a dedication to the Muses.[2]
-                     The poem is traditionally published with the works of Balbilla, though the
-                     internal evidence suggests a different author.[1]
-                    In the poem, Demo explains that Memnon has shown her special respect. In
-                     return, Demo offers the gift for poetry, as a gift to the hero. At the end
-                     of this epigram, she addresses Memnon, highlighting his divine status by
-                     recalling his strength and holiness.[2]
-                    Demo, like Julia Balbilla, writes in the artificial and poetic Aeolic
-                     dialect. The language indicates she was knowledgeable in Homeric
-                     poetry—'bearing a pleasant gift', for example, alludes to the use of that
-                     phrase throughout the Iliad and Odyssey.[a][2];
-                """;
-
-        CompletionSKContext context = summarize.invokeAsync(text).block();
+        CompletionSKFunction summarize = kernel.getSemanticFunctionBuilder()
+                .createFunction(
+                        prompt,
+                        "summarize",
+                        null,
+                        null,
+                        new PromptTemplateConfig.CompletionConfig(
+                                0.2, 0.5, 0, 0, 2000, new ArrayList<>()));
+
+        if (summarize == null) {
+            LOGGER.error("Null function");
+            return;
+        }
 
-        if (context != null) {
-            LOGGER.info("Result: " + context.getResult());
-        } else {
-            LOGGER.error("Null result");
+        summarize.invokeAsync(TEXT_TO_SUMMARIZE).subscribe(
+                context -> {
+                    LOGGER.info("Result: {} ", context.getResult());
+                },
+                error -> {
+                    LOGGER.error("Error: {} ", error.getMessage());
+                },
+                () -> {
+                    LOGGER.info("Completed");
+                });
+
+        try {
+            TimeUnit.SECONDS.sleep(10);
```

Comment:
The 10 second sleep does make this a bit probabilistic, especially when we have throttles on usage. I would use a count down latch to block externally. i.e

```
CountDownLatch cdl = new CountDownLatch(1);
...
() -> {
    LOGGER.info("Completed");
    cdl.countDown();
})

cdl.await();
```